extensions/smallrye-reactive-messaging-kafka/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/SmallRyeReactiveMessagingKafkaProcessor.java

```diff
@@ -532,4 +548,113 @@ private String serializerDeserializerFor(DefaultSerdeDiscoveryState discovery, T
         // unknown
         return null;
     }
+
+    @BuildStep
+    @Consume(RuntimeConfigSetupCompleteBuildItem.class)
+    public void reflectiveValueSerializerPayload(CombinedIndexBuildItem combinedIndex,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        IndexView index = combinedIndex.getIndex();
+        Config config = ConfigProvider.getConfig();
+
+        processOutgoingForReflectiveClassPayload(index, config,
+
+                (annotation, payloadType) -> produceReflectiveClass(reflectiveClass, annotation, payloadType));
+
+        processOutgoingChannelForReflectiveClassPayload(index, config,
+                (annotation, payloadType) -> produceReflectiveClass(reflectiveClass, annotation, payloadType));
+
+        processIncomingForReflectiveClassPayload(index, config,
+                (annotation, payloadType) -> produceReflectiveClass(reflectiveClass, annotation, payloadType));
+
+        processIncomingChannelForReflectiveClassPayload(index, config,
+                (annotation, payloadType) -> produceReflectiveClass(reflectiveClass, annotation, payloadType));
+
+    }
+
+    void produceReflectiveClass(BuildProducer<ReflectiveClassBuildItem> reflectiveClass, AnnotationInstance annotation,
```

Comment:
The `annotation` parameter is useless?

Reply:
I needed it while testing but then changed how to call in tests. I'll remove it.